public/app/plugins/panel/table/TableCellOptionEditor.tsx

```diff
@@ -64,6 +64,7 @@ export const TableCellOptionEditor = ({ value, onChange }: Props) => {
       // Set the new type of cell starting
       // with default settings
       value = {
+        // @ts-ignore
```

Comment:
Seems like a bit of bummer to lose typing here, probably worth a follow up but I think the select can be directly changed to work with `TableCellOptions` instead of just the type. 

I was able to get typescript to not complain using these types:

```typescript
interface CellDisplayModeOption {
  value: TableCellOptions;
  label: string
}

const cellDisplayModeOptions: CellDisplayModeOption[] = [
  { value: { type: TableCellDisplayMode.Auto } , label: 'Auto' },
  { value: { type: TableCellDisplayMode.ColorText }, label: 'Colored text' },
  { value: { type: TableCellDisplayMode.ColorBackground }, label: 'Colored background' },
  { value: { type: TableCellDisplayMode.Gauge }, label: 'Gauge' },
  { value: { type: TableCellDisplayMode.JSONView }, label: 'JSON View' },
  { value: { type: TableCellDisplayMode.Image }, label: 'Image' },
];
```

Don't think that should prevent merging this but perhaps worth following up with.